### PR TITLE
Fixing exhaustiveness warnings.

### PR DESCRIPTION
**Pull request by paulp**
ClassfileAnnotArg becoming sealed brought tidings of not being
exhaustive.

### CODE REVIEW

**Comment by soc** [comment]
Thanks!

**Comment by adriaanm** [comment]
LGTM!

File: src/compiler/scala/reflect/reify/codegen/GenAnnotationInfos.scala [hunk 1]
```diff
@@ -45,6 +45,8 @@ trait GenAnnotationInfos {
         mirrorFactoryCall(nme.ArrayAnnotArg, scalaFactoryCall(nme.Array, args map reifyClassfileAnnotArg: _*))
       case NestedAnnotArg(ann) =>
         mirrorFactoryCall(nme.NestedAnnotArg, reifyAnnotationInfo(ann))
+      case _ =>
+        sys.error(s"Don't know what to do with $arg")
```
Comment (xeno-by) [inline src/compiler/scala/reflect/reify/codegen/GenAnnotationInfos.scala:49]: Why is the treatment here different from the rest of the commit?

**Comment by xeno-by** [comment]
Also if we check everything to `@unchecked`, then what is the benefit of having `ClassfileAnnotArg` sealed?

**Comment by paulp** [comment]
You didn't see me sealing it. If you have a better plan for those types, then be my guest. But it's irresponsible to merge in a bunch of build noise and then just leave it indefinitely.

**Comment by xeno-by** [comment]
How about unsealing it?

**Comment by magarciaEPFL** [comment]
LGTM

**Comment by adriaanm** [comment]
When you need to add a subclass outside of the compilation unit, feel free to remove the sealed keyword. Until then, it's a good way to improve navigability etc.

File: src/compiler/scala/reflect/reify/codegen/GenAnnotationInfos.scala (hunk 1 above)
Comment (adriaanm) [inline src/compiler/scala/reflect/reify/codegen/GenAnnotationInfos.scala:49]: I don't see the problem of generating a nicer error message here. The others just get a matcherror.

**Pull request merged**